app/models/pipeline_run.rb

```diff
@@ -1445,10 +1445,15 @@ def get_summary_contig_counts(min_contig_size)
 
   def get_summary_contig_counts_v2(min_contig_size)
     summary_dict = {} # key: count_type:taxid , value: contigs, contig_reads
-    contig_lineages(min_contig_size).each do |c|
-      lineage = JSON.parse(c.lineage_json)
-      lineage.each do |count_type, taxid_arr|
-        taxids = taxid_arr[0..1]
+    contig_taxids = contigs.select("read_count, species_taxid_nt, species_taxid_nr, genus_taxid_nt, genus_taxid_nr")
+                           .where("read_count >= ?", min_contig_size)
+                           .where("lineage_json IS NOT NULL")
+    contig_taxids.each do |c|
+      count_types_to_taxids = {
```

Comment:
This construct seem unnecessary... we should be able to do something like

```
# before the loop, to always initialize new keys with 0
summary_per_tax = Hash.new(0) 

# in the loop
read_count, species_taxid_nt, species_taxid_nr, genus_taxid_nt, genus_taxid_nr = c

summary_dict[species_taxid_nt]["nt"][read_count] += 1 if species_taxid_nt
summary_dict[species_taxid_nr]["nr"][read_count] += 1 if species_taxid_nr

```

Reply:
Done (initializing to `Hash.new(0)` became a longer code block to ensure the default values would actually be [stored in the hash](https://ruby-doc.org/core-2.5.1/Hash.html#method-c-new) and rubocop enforced some styling).